src/vm/collection.rs

```diff
@@ -139,4 +139,21 @@ pub trait Collection<VM: VMBinding> {
         // By default, MMTk assumes the amount of memory the VM allocates off-heap is negligible.
         0
     }
+
+    /// Callback function to ask the VM whether GC is disabled, disallowing MMTk to trigger garbage
+    /// collection. When collection is disabled, you can still allocate through MMTk, but MMTk will
+    /// not trigger a GC even if the heap is full. In such a case, the allocation will exceed MMTk's
+    /// heap size (the soft heap limit). However, there is no guarantee that the physical allocation
+    /// will succeed, and if it succeeds, there is no guarantee that further allocation will keep
+    /// succeeding. So if a VM disables collection, it needs to allocate with careful consideration
+    /// to make sure that the physical memory allows the amount of allocation. We highly recommend
+    /// to have GC always enabled (i.e. that this method always returns false). However, we support
+    /// this to accomodate some VMs that require this behavior. Note that this call also disables
+    /// explicit GCs (through handle_user_collection_request()). Note also that any synchronization
+    /// involving enabling and disabling collections by mutator threads should be implemented by the VM.
+    fn is_collection_disabled() -> bool {
```

Comment:
I feel `is_collection_enabled()` more intuitive because otherwise I need to deal with the case of GC "not disabled" which is a double negation.

p.s. Although Chinese love double negation (while English seem to hate double negation), it is still a bit mind-bending when I see double negation in the API.  I tried to implement this for mmtk-ruby, and I got it wrong on my first attempt.